Framework/Core/include/Framework/HistogramRegistry.h

```diff
@@ -345,7 +345,7 @@ auto& HistogramRegistry::operator()(const HistName& histName)
 }
 
 template <typename T>
-void HistogramRegistry::insertClone(const HistName& histName, const std::shared_ptr<T>& originalHist)
+HistPtr HistogramRegistry::insertClone(const HistName& histName, const std::shared_ptr<T>& originalHist)
```

Comment:
This function is able also to clone whole directory structures. 
The pointer that will be returned here in this case would therefore be unpredictable (it will be the last histogram in the registry that matches the request).
My suggestion would be to keep this as a void function for now and later we can split it into a version for cloning single histograms that returns the corresponding pointer and a version that we call `addGroupClone` or similar which will then be a `void` function.